### PR TITLE
Replace uses of the PYTHON_RUNTIMES env var by explicit parameter

### PR DESCRIPTION
**Pull request by KSerrania**
### What does this PR do?

Makes invoke functions take the python runtimes list as an explicit parameter rather than an environment variable.
The `PYTHON_RUNTIMES` environment variable remains in the `.gitlab-ci.yml` file as it's used when multiple build jobs are using the same base build job, to pass the specific job parameters.

### Motivation

Clarify how the builds are configured.

#### Windows docker build

The windows docker build still passes the arguments as environment variables. The reason why I'm doing this is that the python runtimes variable can contain commas, which are considered parameter delimiters in cmd and powershell. Thus, doing `\build-agent6.bat $RELEASE_VERSION $AGENT_MAJOR_VERSION $PYTHON_RUNTIMES` with `PYTHON_RUNTIMES=2,3` results in two separate args in the batch script (`%3=2` and `%4=3`). The correct way to do this would be to surround `PYTHON_RUNTIMES` with double quotes, but we didn't find a way to do it, as the command is parsed by the gitlab yaml parser + powershell inside the runner + docker + cmd inside the docker container which makes escaping the double quotes difficult.

An alternative solution would be to change the delimiter we use for the python runtimes to one that is not a delimiter for powershell, cmd or bash.

#### Python build flags

Something strange I found when doing this PR is that some invoke functions I expected not to use the `PYTHON_RUNTIMES` variable did use it, for instance:
- `inv process-agent.build`
- `inv system-probe.build`
- `inv trace-agent.build`
They need the variable because they call `get_build_flags`, which needs the python runtimes as it contains this line:
```
ldflags += "-X {}/pkg/config.DefaultPython={} ".format(REPO_PATH, get_default_python(python_runtimes))
```
However, as far as I know, this ldflag is useless for process / system-probe / trace (and I think that's not the only useless ldflag added in the method). Should we amend `get_build_flags`, for instance add a `python_flags` parameter which would be set to false for these builds so that it skips python-related flags? Or maybe we should split it in multiple functions? 

#### Alternative / Future solution (should probably be discussed separately with an RFC)

An alternative method of dealing with this would be to remove the use of the `PYTHON_RUNTIMES` variables entirely, and rely only on the major version (with an agent version -> python runtimes map defined somewhere).

Advantages:
- Less config parameters to keep track of
- Up to now, we have never made a 6 build (since 6.13.0) that does not contain both Py2 and Py3, nor we have made a 7 build that contains something else than Py3.

Drawbacks / Questions:
- Less configurable builds (though, for a custom build, you could probably amend the mapping between Agent versions and python runtimes)
- Some source tests only use Py2. How do we deal with them? Will they still work as expected if we run them in `6` mode (Py2+Py3), or will we have to find a workaround?

### CODE REVIEW

**Review by albertvaka — DISMISSED** [review]
It's hard to tell if the changes are correct just looking at the code, but I didn't see anything obviously wrong. Let's merge and see 👍

**Comment by albertvaka** [comment]
+1 on not making `get_build_flags` not need `PYTHON_RUNTIMES`.

**Comment by albertvaka** [comment]
+1 on moving from `PYTHON_RUNTIMES` to major version. Also, on the long run, it could allow for further differences between the agents.

**Review by albertvaka — DISMISSED** [review]
LGTM

File: tasks/utils.py [hunk 1]
```diff
@@ -50,9 +50,18 @@ def get_multi_python_location(embedded_path=None, rtloader_root=None):
...
     return rtloader_lib, rtloader_headers, rtloader_common_headers
 
+def has_both_python(python_runtimes):
+    python_runtimes = python_runtimes.split(',')
+    return '2' in python_runtimes and '3' in python_runtimes
+
+def get_win_py_runtime_var(python_runtimes):
+    python_runtimes = python_runtimes.split(',')
+
+    return "PY2_RUNTIME" if '2' in python_runtimes else "PY3_RUNTIME"
```
Comment (truthbk) [inline tasks/utils.py]: Probably a matter of discussion for another PR, but this magic string variable thing is a little nasty.

File: tasks/utils.py [hunk 2]
```diff
@@ -168,14 +174,13 @@ def get_git_commit():
...
     return check_output(['git', 'rev-parse', '--short', 'HEAD']).decode('utf-8').strip()
 
-def get_default_python():
+def get_default_python(python_runtimes):
     """
     Get the default python for the current build:
-    - default to 3 if PYTHON_RUNTIMES is not specified (so that dev builds default to 3)
-    - default to 2 if PYTHON_RUNTIMES includes both versions (so that builds with 2 and 3 default to 2)
+    - default to 2 if python_runtimes includes 2 (so that builds with 2 and 3 default to 2)
+    - default to 3 if python_runtimes includes only 3
```
Comment (truthbk) [inline tasks/utils.py]: nit: in actuality it will default to '3'  if '2' not in `python_runtimes`, no matter what it is.

**Review by truthbk — DISMISSED** [review]
Looks good! Just a couple of tiny nits; minor and ignorable.

**Review by albertvaka — APPROVED** [review]

**Review by truthbk — APPROVED** [review]

**Pull request merged**